### PR TITLE
issue #403 - Upgrade to FHIR 4.0.1

### PR DESCRIPTION
**Pull request by lmsurpre**
Added 4.0.1 definitions and updated CodeGenerator and model
* Resource.id is now a java.lang.String and not an Id (still with id format constraint, but no extensions allowed)
  * Updated PathAwareVisitor to track the path as we visit non-value elements of type java.lang.String (i.e. ones like `Resource.id` but not ones like `Resource.Meta.versionId.value`)
* Removed unneccessary escaping of keywords in generated constraint annotations 

Updated fhir-examples
* fixed generator issues and regenerated all ibm examples
* downloaded spec from https://www.hl7.org/fhir/downloads.html and overwrote the 4.0.0 examples
* updated and reorganized all index files

Updated Validator and FHIRPathEvaluator to support the new `%rootResource` external constant

Signed-off-by: Lee Surprenant <lmsurpre@us.ibm.com>

### CODE REVIEW

File: fhir-model/src/main/java/com/ibm/fhir/model/resource/AppointmentResponse.java [hunk 1]
```diff
@@ -625,7 +624,7 @@ public Builder actor(Reference actor) {
          * <p>This element is required.
          * 
          * @param participantStatus
-         *     accepted | declined | tentative | in-process | completed | needs-action | entered-in-error
```
Comment (lmsurpre) [inline fhir-model/src/main/java/com/ibm/fhir/model/resource/AppointmentResponse.java]: note:  this is a breaking change!

File: fhir-model/src/main/java/com/ibm/fhir/model/resource/CarePlan.java [hunk 2]
```diff
@@ -994,7 +993,7 @@ public Builder partOf(Collection<Reference> partOf) {
          * <p>This element is required.
          * 
          * @param status
-         *     draft | active | suspended | completed | entered-in-error | cancelled | unknown
```
Comment (lmsurpre) [inline fhir-model/src/main/java/com/ibm/fhir/model/resource/CarePlan.java]: this is a breaking change!

File: fhir-examples/src/main/resources/json/ibm/complete-absent/Account-1.json [hunk 3]
```diff
@@ -1,13 +1,5 @@
 {
     "resourceType": "Account",
-    "_id": {
-        "extension": [
-            {
-                "url": "http://hl7.org/fhir/StructureDefinition/data-absent-reason",
```
Comment (prb112) [inline fhir-examples/src/main/resources/json/ibm/complete-absent/Account-1.json]: I take it this is no longer used? it's absent = therefore it should be treated as absent?

Comment (prb112) [reply fhir-examples/src/main/resources/json/ibm/complete-absent/Account-1.json]: Crazy question, what happens when you leave it?  If it'd be valid, maybe we should have a test with data like this?

Comment (lmsurpre) [reply fhir-examples/src/main/resources/json/ibm/complete-absent/Account-1.json]: as of 4.0.1 that is invalid.  i did manually test this and it complains its an unrecognized element (which i think is the right behavior), but didn't introduce a negative test case for it.

**Pull request merged**